lib/_tls_wrap.js

```diff
@@ -1256,6 +1256,11 @@ exports.connect = function connect(...args) {
       localAddress: options.localAddress,
       lookup: options.lookup
     };
+
+    if (options.timeout) {
```

Comment:
`typeof options.timeout === 'number'` to allow setting a timeout of `0`.

Reply:
0 == disabled idle timeout so I'm not sure if it's very useful.

Reply:
This code for tls.connect() is identical to that in net.connect(), https://github.com/nodejs/node/blob/2c7f4f474bfbb19b7ae6597112cca41141bf71a4/lib/net.js#L158-L160, which is what I'd expect, so LGTM as is.